errors/errors.go

```diff
@@ -0,0 +1,99 @@
+package zqe
+
+import (
+	"bytes"
+	"fmt"
+	"runtime"
+)
+
+type Kind int
+
+const (
+	Other Kind = iota
+	Invalid
+	NotFound
+	Exists
+	Conflict
+)
+
+func (k Kind) String() string {
+	switch k {
+	case Other:
+		return "other error"
+	case Invalid:
+		return "invalid operation"
+	case NotFound:
+		return "item does not exist"
+	case Exists:
+		return "item already exists"
+	case Conflict:
+		return "conflict with pending operation"
+	}
+	return "unknown error kind"
+}
+
+type Error struct {
+	Kind Kind
+	Err  error
+}
+
+func pad(b *bytes.Buffer, s string) {
+	if b.Len() == 0 {
+		return
+	}
+	b.WriteString(s)
+}
+
+func (e *Error) Error() string {
+	b := &bytes.Buffer{}
+	if e.Kind != Other {
+		pad(b, ": ")
```

Comment:
Is this `pad()` needed?

Reply:
pad() checks if b.len() is zero, so it's to keep in.